packages/web3-eth/test/integration/defaults.test.ts

```diff
@@ -0,0 +1,525 @@
+	getTransactionFromAttr,
+	getTransactionType,
+} from '../../src/utils';
+import { BasicAbi, BasicBytecode } from '../shared_fixtures/build/Basic';
+import { detectTransactionType } from '../../dist';
+import { getTransactionGasPricing } from '../../src/utils/get_transaction_gas_pricing';
+
+describe('defaults', () => {
+	let web3Eth: Web3Eth;
+	let accounts: string[] = [];
+	let clientUrl: string;
+	let contract: Contract<typeof BasicAbi>;
+	let deployOptions: Record<string, unknown>;
+	let sendOptions: Record<string, unknown>;
+
+	beforeAll(async () => {
+		clientUrl = getSystemTestProvider();
+		const acc1 = await createNewAccount({ unlock: true, refill: true });
+		const acc2 = await createNewAccount({ unlock: true, refill: true });
+		accounts = [acc1.address, acc2.address];
+		web3Eth = new Web3Eth(clientUrl);
+
+		contract = new Contract(BasicAbi, undefined, undefined, web3Eth.getContextObject() as any);
+
+		deployOptions = {
+			data: BasicBytecode,
+			arguments: [10, 'string init value'],
+		};
+
+		sendOptions = { from: accounts[0], gas: '1000000' };
+
+		contract = await contract.deploy(deployOptions).send(sendOptions);
+	});
+	afterAll(() => {
+		if (clientUrl.startsWith('ws')) {
+			(web3Eth.provider as WebSocketProvider).disconnect();
+		}
+	});
+
+	describe('defaults', () => {
+		it('defaultAccount', () => {
+			// default
+			expect(web3Eth.defaultAccount).toBeNull();
+
+			// after set
+			web3Eth.setConfig({
+				defaultAccount: accounts[0],
+			});
+			expect(web3Eth.defaultAccount).toBe(accounts[0]);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultAccount: accounts[1],
+				},
+			});
+			expect(eth2.defaultAccount).toBe(accounts[1]);
+
+			// check utils
+			expect(getTransactionFromAttr(eth2)).toBe(accounts[1]);
+		});
+		it('handleRevert', () => {
+			// default
+			expect(web3Eth.handleRevert).toBe(false);
+
+			// after set
+			web3Eth.setConfig({
+				handleRevert: true,
+			});
+			expect(web3Eth.handleRevert).toBe(true);
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					handleRevert: true,
+				},
+			});
+			expect(eth2.handleRevert).toBe(true);
+		});
+		it('defaultBlock', async () => {
+			// default
+			expect(web3Eth.defaultBlock).toBe('latest');
+
+			// after set
+			web3Eth.setConfig({
+				defaultBlock: 'earliest',
+			});
+			expect(web3Eth.defaultBlock).toBe('earliest');
+
+			// set by create new instance
+			const eth2 = new Web3Eth({
+				provider: clientUrl,
+				config: {
+					defaultBlock: 'earliest',
+				},
+			});
+			expect(eth2.defaultBlock).toBe('earliest');
+			const acc = await createNewAccount({ refill: true });
+			// check implementation
+			const balance = await eth2.getBalance(acc.address);
+			expect(balance).toBe('0x0');
+			eth2.setConfig({
+				defaultBlock: 'latest',
+			});
+			const balanceLatest = await eth2.getBalance(acc.address);
```

Comment:
we need to add test for checking: if explicitly passing blockNum overrides defaultBlock